build failed with WITH_GPU=ON and WITH_TESTING=ON

```bash
libsendrecvop_grpc.a(sendrecvop_utils.cc.o): In function `paddle::operators::detail::SerializeToByteBuffer(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, paddle::framework::Variable*, paddle::platform::DeviceContext const&, grpc::ByteBuffer*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda(void*)#2}::_FUN(void*)':
/paddle/paddle/fluid/operators/detail/sendrecvop_utils.cc:156: undefined reference to `void paddle::memory::Free<paddle::platform::CUDAPinnedPlace>(paddle::platform::CUDAPinnedPlace, void*)'
collect2: error: ld returned 1 exit status
```

--- a/paddle/fluid/operators/detail/sendrecvop_utils.cc
+++ b/paddle/fluid/operators/detail/sendrecvop_utils.cc
@@ -149,12 +149,14 @@ void SerializeToByteBuffer(const std::string& name, framework::Variable* var,
   }
 
   if (platform::is_gpu_place(ctx.GetPlace())) {
+#ifdef PADDLE_WITH_CUDA
     // GPU data is copied to CPU buffer when sending,
     // free the buffer when possible.
     destroy_callback = [](void* backing) {
       platform::CUDAPinnedPlace cuda_pinned;
       memory::Free(cuda_pinned, backing);
     };
+#endif
   }
 
   std::string header;

